test-data/unit/semanal-errors.test

```diff
@@ -1056,10 +1056,19 @@ from typing import Generic as t # E: Name 't' already defined on line 2
 def f(x: 'foo'): pass # E: Name 'foo' is not defined
 [out]
 
-[case testInvalidStrLiteralType2]
-def f(x: 'int['): pass # E: syntax error in type comment
+[case testInvalidStrLiteralStrayBrace]
+def f(x: 'int['): pass # E: Invalid type. Try using Literal['int['] instead?
```

Comment:
What about showing this suggestion only if the string parses with an error *and* contains a square bracket?

I think it would be nice to show the "Try using Literal[...]" note if the user tries doing something like `def f(x: "some-id-here")` -- but we would also get a parse error if we try parsing that string. I'm thinking that a nice tradeoff might be to assume that any string containing an opening or closing square bracket is meant to probably be a type.

Reply:
`some-id-here` is a valid Python expression.

Reply:
I think that suggesting literals types if there is a string literal that is not a valid type is questionable, since it's quite likely that the user was just trying (and failing) to use a string literal escape. Any heuristic that we may come up with is going to be just a guess since we don't know how users will actually use literal types. Since this clearly not a straightforward thing, please leave the logic out from this initial PR and we can consider getting back to it at some point. It doesn't seem worth it to slow down progress because of this issue.